test/parallel/test-vm-context.js

```diff
@@ -62,9 +62,27 @@ assert.ok(gh1140Exception,
           'expected exception from runInContext signature test');
 
 // GH-558, non-context argument segfaults / raises assertion
-[undefined, null, 0, 0.0, '', {}, []].forEach(function(e) {
-  assert.throws(function() { script.runInContext(e); }, TypeError);
-  assert.throws(function() { vm.runInContext('', e); }, TypeError);
+const nonContextualSandboxErrorMsg =
+  /^TypeError: contextifiedSandbox argument must be an object\.$/;
+
+[undefined, null, 0, 0.0, ''].forEach(function(e) {
+
+  assert.throws(function() { script.runInContext(e); },
+                nonContextualSandboxErrorMsg);
+  assert.throws(function() { vm.runInContext('', e); },
+                nonContextualSandboxErrorMsg);
+});
+
+// GH-558, non-context argument segfaults / raises assertion
+const contextifiedSandboxErrorMsg =
+  /^TypeError: sandbox argument must have been converted to a context\.$/;
+
+[{}, []].forEach(function(e) {
+
```

Comment:
@cjihrig that was my original thought, but i figured I'd skip the type checking and be more direct. I still have one more PR for this file and could change it then. What do you recommend?